### PR TITLE
Mesos CNI integration tests

### PR DESCRIPTION
**Pull request by rchirakk**
Mesos tests and fixes in endpoint inspection.

### CODE REVIEW

File: netctl/netctl.go [hunk 1]
```diff
@@ -619,7 +619,9 @@ func createEndpointGroup(ctx *cli.Context) {
 }
 
 func inspectEndpointGroup(ctx *cli.Context) {
-	argCheck(1, ctx)
+	if len(ctx.Args()) != 1 {
+		errExit(ctx, exitHelp, "endpoint group name is required", true)
+	}
```
Comment (gaurav-dalvi) [inline netctl/netctl.go]: Please remove this change. I have another PR https://github.com/contiv/netplugin/pull/539 
 for this and some other stuff.

File: netmaster/objApi/apiController.go [hunk 2]
```diff
@@ -407,7 +407,10 @@ func (ac *APIController) EndpointGetOper(endpoint *contivModel.EndpointInspect)
 	if err == nil {
 		for _, epCfg := range epCfgs {
 			ep := epCfg.(*mastercfg.CfgEndpointState)
-			if strings.Contains(ep.ContainerID, endpoint.Oper.Key) || strings.Contains(ep.ContainerName, endpoint.Oper.Key) {
+			if strings.Contains(ep.EndpointID, endpoint.Oper.Key) ||
```
Comment (gaurav-dalvi) [inline netmaster/objApi/apiController.go]: What is the logic here ? Just for my understanding.

Comment (rchirakk) [inline netmaster/objApi/apiController.go]: endpoint inspect to check matching endpoint-id.

File: netctl/netctl.go (hunk 1 above)
Comment (rchirakk) [inline netctl/netctl.go]: sure.

**Comment by rchirakk** [comment]
@shaleman @jainvipin  PTAL

File: test/integration/mesos_test.go [hunk 3]
```diff
@@ -0,0 +1,843 @@
...
+	err = docker.RemoveContainer(cinfo.CniContainerid, true, true)
+	assertNoErr(err, c, fmt.Sprintf("remove container %s", cinfo.CniContainerid))
+	delete(containerInfo, cfg1.containerName)
+}
+
+func cleanupContainers() {
+	intLog.Infof("######### cleaning containers ########")
+	for containerName := range containerInfo {
+		intLog.Infof("cleanup container %s", containerName)
+		docker, _ := dockerclient.NewDockerClient("http://localhost:2385", nil)
```
Comment (jainvipin) [inline test/integration/mesos_test.go]: do we need this for every loop?

File: test/integration/mesos_test.go [hunk 4]
```diff
@@ -0,0 +1,843 @@
...
+		return
+	}
+
+	containerName := cfg1.containerName
+	intLog.Infof("creating container: %s", containerName)
+
+	docker, _ := dockerclient.NewDockerClient("http://localhost:2385", nil)
+
+	// Create a container
+	docker.PullImage("alpine", nil)
```
Comment (jainvipin) [inline test/integration/mesos_test.go]: let's check the return value of this

File: test/integration/mesos_test.go [hunk 5]
```diff
@@ -0,0 +1,843 @@
...
+	}
+
+	// check if container exists
+	if _, ok := containerInfo[cfg1.containerName]; !ok {
+		intLog.Warnf("no conainer with name %s", cfg1.containerName)
+		return
+	}
+
+	cinfo := &cfg1.reqAttr
+	docker, _ := dockerclient.NewDockerClient("http://localhost:2385", nil)
```
Comment (jainvipin) [inline test/integration/mesos_test.go]: should we ignore the error?

**Pull request merged**